Beat & Note Behavior

Bretterbear opened the pull request:
Functional updates to beatspawning & detection including
1. random note spawning mode for input testing
2. read from file mode for playing back levels
3. Note hits now have the ability to detect & discriminate between degrees of "good"
4. Note Assembly Prefab laid out a bit more nicely w/ awesome placeholder art (courtesy of Charli!)

Bretterbear committed:
Random Beat Spawning

Created a random spawn beat system that'll just spawn random beats on up/down/left/right locales. NOTE - YOU NOW HAVE TO USE THE BEATSPAWNER PREFAB AS IT HAS THE CORRECT REFERENCES LOADED.

Bretterbear committed:
Merge branch 'main' into PlayerInput

Bretterbear committed:
BeatSpawner Update

Add ability to read from file for beatspawning patterns. Also updated the random function to be able to test multi-input

Bretterbear committed:
Note Update

Changes to Note & BeatSystem prefabs to use tmp art assets & to be aligned properly!

Swiftshadow commented:
Overall looks good! There are a few places that can be condensed down a bit though:

General:
 - Make sure all canvases are "Scale with Screen Size"

BeatSpawner.cs:
- General change: Change the enum values so down is 0, right is 1, up 2, and left 3. This will help condense code later.
- Spawn point transforms into an array (ex. Spawnpoints) instead of discrete variables for future use. Above new mapping followed
- Lines 76-102 (The ReadSpawn method): The if statements can be turned into one function with a string parameter and called 4 times. 
    - (Optional, this is more style-based than functional) I can do some transformations to the data on the reader end (mapping the string name to an int) and it can be further condensed into a for loop with one call to the new function.
 - Lines 130-134 updated to the new enum mapping
 - Lines 147-167 can be replaced entirely by:
     - int noteStyleInt = (int)noteStyle // Optional line, where this var is used could be two separate casts
     - noteObject.transform.position = Spawnpoints[noteStyleInt]
     - noteObject.transform.Rotate(new Vector3(0, 0, noteStyleInt * 90);
 - The above works because enums can be converted 1-1 to an int, and the spawning follows regular patterns
 
NoteObject.cs:
- What unit is the gradingThreshold in? A [Tooltip("")] comment above it will explain that in code and when hovered over in the Inspector

Bretterbear committed:
Modification to Pull Request

Small tweaks
-UI scaling mode fixed
-changes to enum useage in beatspawner
-note object documentation improved
